rbac/management/principal/cleaner.py

```diff
@@ -199,28 +205,24 @@ def process_umb_event(frame, umb_client: Stomp, bootstrap_service: TenantBootstr
             logger.info("process_umb_event: Another listener is running. Aborting.")
             return False
 
-        data_dict = xmltodict.parse(frame.body)
-        canonical_message = data_dict.get("CanonicalMessage")
-        if canonical_message:
-            try:
-                user = retrieve_user_info(canonical_message)
-            except Exception as e:  # Skip processing and leave the it to be processed later
-                logger.error("process_umb_event: Error retrieving user info: %s", str(e))
-                return True
+        try:
+            data_dict = xmltodict.parse(frame.body)
+            canonical_message = data_dict.get("CanonicalMessage")
 
+            user = retrieve_user_info(canonical_message)
             # By default, only process disabled users.
             # If the setting is enabled, process all users.
             if not user.is_active or settings.PRINCIPAL_CLEANUP_UPDATE_ENABLED_UMB:
                 # If Tenant is not already ready, don't ready it
                 bootstrap_service.update_user(user, ready_tenant=False)
-        else:
-            # Message is malformed.
-            # Ensure we dont block the entire queue by discarding it.
-            # TODO: this is not the only way a message can be malformed
-            pass
+            umb_client.ack(frame)
+            stomp_messages_ack_total.inc()
+        except Exception as e:
```

Comment:
We may want to introduce two types of exceptions:

1. Those that are retryable
2. Those that aren't

If we think about categorizing the problems in terms of HTTP status codes, 1 would be 5xx errors, and 2 would be 4xx errors.

This is similar to how KafkaConnect deals with exceptions also.

I say this because of broker retry behavior may not always be optimized for both cases (as in our case).

For now I can think of two ways forward:

1. Keep as is and see how it works out in practice. Best case temporary failures are successfully retried and we don't have to intervene. Worst case bad messages pause the queue for a while while the broker retries, impacting UX.
2. Use ack instead of nack here, and manually retry failures. Best case bad messages never hold up the queue, however worst case we have a lot of missed messages due to transient errors.

Either way you go, we will I think want to monitor and based on experience categorize the exceptions so we can ack vs nack accordingly eventually.

Reply:
Maybe something we could try for now would be like

```
        except OSError as e:
            # log as error & nack for retry...
        except Exception as e:
            # log as critical & ack so not retried
```

(but up to you, just doing some research on python exception hierarchy :))

Reply:
Cool, yea, a bit leaning towards the first option because the number of missed messages might be large, would be hard to manually retry all the failures. Better late then never, lol.

Yea, as you mentioned, we can iterate on the except / nack / ack handling piece later, I created a JIRA task to https://issues.redhat.com/browse/RHCLOUD-36806